src/components/States.js

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useDispatch, useSelector } from "react-redux";
+import * as entitiesActions from "../store/entities";
+import City from "./City";
+import useFilter from "../shared/useFilter";
+
+const States = ({ countryName, userInput }) => {
+  const dispatch = useDispatch();
+  const country = useSelector((state) =>
+    state.entities.countries.find((country) => country.name === countryName)
+  );
+
+  const filteredStates = useFilter({
+    searchItem: userInput,
+    dataset: country?.states,
+    searchCriteria: "name",
+  });
```

Comment:
- [ ] **OPTIONAL**
Greatly done putting filtering logic to each country 👏 However, there's a small issue that'd be great if you managed to fix it. 😉 💯 
When I search for a city within a country, it works fine, but when I get its weather data, the filtering resets, then I have to scroll down or search for it again to see the results! 🙂